assembly/src/lib.rs

```diff
@@ -234,15 +239,19 @@ impl AsRef<str> for LibraryNamespace {
 }
 
 impl Serializable for LibraryNamespace {
-    fn write_into(&self, target: &mut ByteWriter) -> Result<(), SerializationError> {
-        target.write_str(&self.name)
+    fn write_into<W: ByteWriter>(&self, target: &mut W) {
+        target.write_u64(self.name.len() as u64);
+        target.write_bytes(self.name.as_bytes());
     }
```

Comment:
I think `u64` is too much - let's use `u16` and add TODOs to enforce this restriction via the constructor.